pkg/tests/ossm/smoke_test.go

```diff
@@ -19,50 +19,195 @@ import (
+			t.Log("Jira related: https://issues.redhat.com/browse/OSSM-3586")
+			t.Log("From proxy yaml, verify the time between status.containerStatuses.state.running.startedAt and status.conditions[type=Ready].lastTransitionTime")
+			ratingPod := pod.MatchingSelector("app=ratings", ns)(t, oc.DefaultOC)
+			ratingYaml := oc.GetYaml(t, ns, "pod", ratingPod.Name)
+			t.Log("Validate from YAML, the proxy startup time is less than 10 seconds for ratings pod")
+			validateStartUpProxyTime(t, ratingYaml)
 		})
 
-		DeployControlPlane(t)
-
-		t.LogStep("Install bookinfo pods with sidecar and sleep pod without sidecar")
-		app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
-
-		t.LogStep("Check whether sidecar is injected in all bookinfo pods")
-		assertSidecarInjectedInAllBookinfoPods(t, ns)
-
-		t.LogStep("Check if bookinfo productpage is running through the Proxy")
-		oc.Exec(t,
-			pod.MatchingSelector("app=sleep", ns), "sleep",
-			"curl -sI http://productpage:9080",
-			assert.OutputContains(
-				"HTTP/1.1 200 OK",
-				"ProductPage returns 200 OK",
-				"ProductPage didn't return 200 OK"),
-			assert.OutputContains(
-				"server: istio-envoy",
-				"HTTP header 'server: istio-envoy' is present in the response",
-				"HTTP header 'server: istio-envoy' is missing from the response"),
-			assert.OutputContains(
-				"x-envoy-decorator-operation",
-				"HTTP header 'x-envoy-decorator-operation' is present in the response",
-				"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+		t.NewSubTest(fmt.Sprintf("upgrade %s to %s", fromVersion, toVersion)).Run(func(t TestHelper) {
+			t.Logf("This test checks whether SMCP becomes ready after it's upgraded from %s to %s and bookinfo is still working after the upgrade", fromVersion, toVersion)
+
+			t.Cleanup(func() {
+				app.Uninstall(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+			})
+
+			t.LogStepf("Upgrade SMCP from %s to %s", fromVersion, toVersion)
+			assertSMCPDeploysAndIsReady(t, toVersion)
+
+			t.LogStep("Check if bookinfo productpage is running through the Proxy after the upgrade")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=sleep", ns), "sleep",
+					"curl -sI http://productpage:9080",
+					assert.OutputContains(
+						"HTTP/1.1 200 OK",
+						"ProductPage returns 200 OK",
+						"ProductPage didn't return 200 OK"),
+					assert.OutputContains(
+						"server: istio-envoy",
+						"HTTP header 'server: istio-envoy' is present in the response",
+						"HTTP header 'server: istio-envoy' is missing from the response"),
+					assert.OutputContains(
+						"x-envoy-decorator-operation",
+						"HTTP header 'x-envoy-decorator-operation' is present in the response",
+						"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+			})
+
+			t.LogStep("Delete Bookinfo pods to validate proxy is still working after recreation and upgrade")
```

Comment:
Do we also need to check that the dataplane works correctly even if the control plane is offline (e.g. by deleting the control plane and checking that the traffic still flows through the proxy?)

Maybe we can do this in the last subtest, where we delete the SMCP.

Reply:
Makes sense, will aded